src/measure_theory/function/l1_space.lean

```diff
@@ -344,7 +344,11 @@ hf.mono $ eventually_of_forall $ λ x, by simp [real.norm_eq_abs, abs_le, abs_no
 lemma has_finite_integral.min_zero {f : α → ℝ} (hf : has_finite_integral f μ) :
   has_finite_integral (λa, min (f a) 0) μ :=
 hf.mono $ eventually_of_forall $ λ x,
-  by simp [real.norm_eq_abs, abs_le, abs_nonneg, neg_le, neg_le_abs_self]
+begin
+  simp [real.norm_eq_abs, abs_le, abs_nonneg],
+  rw neg_le,
+  apply neg_le_abs_self,
+end
```

Comment:
Why was this needed?

Reply:
I've no idea! The original proof no longer seemed to work for some reason.

Reply:
If I restore the original proof, lean says:
```
error: tactic failed, there are unsolved goals
state:
α : Type u_1,
m : measurable_space α,
μ : measure α,
f : α → ℝ,
hf : has_finite_integral f μ,
x : α
⊢ -min (f x) 0 ≤ |f x|
```

Reply:
Replacing the proof with:
```
simp [real.norm_eq_abs, abs_le, abs_nonneg, neg_le, neg_le_abs_self, abs_eq_max_neg, le_total],
```
Also works, and is closer in spirit to the original proof.  

Before, `abs` was defined (via `to_additive`) as
```
def abs {α : Type*} [has_neg α] [lattice α] (a : α) : α := a ⊔ (-a)
```
whereas now we're relying on ℝ being a linear order with `neg` and therefore a lattice with `neg`, which therefore `has_abs`, and I guess `simp` needs the `abs_eq_max_neg` and `le_total` lemmas to deal with this.

Reply:
I've now changed the proof to:
```
  by simp [real.norm_eq_abs, abs_le, abs_nonneg, neg_le, neg_le_abs_self, abs_eq_max_neg, le_total]
```
so will resolve this. Thanks for helping me to find a better solution.